proposals/000-decorator-api.md

```diff
@@ -0,0 +1,322 @@
+
+For example, decorations can be used to draw highlights, images or buttons.
+
+#### Properties
+
+* `id: String`
+    * An identifier for this decoration. It must be unique in the group the decoration is applied to.
+* `locator: Locator`
+    * Location in the publication where the decoration will be rendered.
+* `style: Decoration.Style`
+    * Declares the look and feel of the decoration.
+* `interaction: Set<Decoration.Interaction>`
+    * List of enabled user interactions for this decoration.
+
+### `Decoration.Interaction` Enum
+
+A type of user interaction with a decoration.
+
+* `Activate`
+    * Primary interaction with a decoration, such as a single tap or click.
+
+The user can interact with a decoration if it is enabled in `decoration.interactions` and supported by the Navigator for its style.
+
+The terminology avoids concrete gestures (e.g. tap) to fit all platforms (mouse vs. touch) and media type. For example, activating a decoration for an Audiobook Navigator might be very different from activating a visual highlight.
+
+### `Decoration.Style` Interface
+
+The Decoration Style determines the look and feel of a decoration once rendered by a Navigator. It is media type agnostic, meaning that each Navigator will translate the style into a set of rendering instructions which makes sense for the resource type.
+
+The Readium toolkit supports two default styles:
+
+* `Highlight(tint: Color?)`
+* `Underline(tint: Color?)`
+
+**Note**: This can be implemented differently depending on the platform capabilities. Ideally, this is a marker interface and each concrete type is used to identify the style.
+
+### Decoration Templates
+
+Decoration Templates translate a `Decoration` object and its associated style into concrete rendering instructions specific to a Navigator and resource type. Each concrete implementation of `DecorableNavigator` can support its own kind of decoration templates.
+
+#### `HTMLDecorationTemplate` Class
+
+An `HTMLDecorationTemplate` renders a `Decoration` into a set of HTML elements and associated stylesheet. It is used to render decorations in EPUB publications, for example.
+
+##### Properties
+
+* `element: (Decoration) -> String`
+    * Closure used to generate a new HTML element for the given `Decoration`. 
+    * Several elements will be created for a single decoration when using the `boxes` layout.
+    * The Navigator will automatically position the created elements according to the decoration's `Locator`. The template is only responsible for the look and feel of the generated elements.
+* `stylesheet: String?`
+    * A CSS stylesheet which will be injected in the resource, which can be referenced by the created elements.
+    * Make sure to use unique identifiers for your classes and IDs to avoid conflicts with the HTML resource itself. Best practice is to prefix with your app name. `r2-` and `readium-` are reserved by the Readium toolkit.
+* `layout: Layout`
```

Comment:
Mentioned by @ettore, we probably need a third type of layout for "continuous boxes bounds" to account for a range spanning two columns.